src/main/java/com/exasol/adapter/properties/PropertyValidator.java

```diff
@@ -0,0 +1,73 @@
+package com.exasol.adapter.properties;
+
+import java.util.Map;
+
+import com.exasol.adapter.AdapterProperties;
+
+/**
+ * Abstract class for validators for adapter properties
+ */
+public class PropertyValidator {
+
+    /**
+     * @return empty chain of property validators, enabling to add arbitrary property validators and validate a given
+     *         map of properties by applying all validators subsequently.
+     */
+    public static ValidatorChain chain() {
+        return new ValidatorChain();
+    }
+
+    /**
+     * @param properties
+     * @return {@code true} if one of the keys of parameter {properties} requires to refresh the virtual schema.
+     */
+    public static boolean requiresRefreshOfVirtualSchema(final Map<String, String> properties) {
+        return properties.containsKey(TableCountLimit.MAXTABLES_PROPERTY)
+                || AdapterProperties.isRefreshingVirtualSchemaRequired(properties);
+    }
```

Comment:
ah, this is the one that disturbed me during my review... The core `PropertyValidator` class should not have to know about any specific properties...

Reply:
OK.